discovery_engine/lib/src/infrastructure/repository/hive_engine_state_repo.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:typed_data' show Uint8List;
+
+import 'package:hive/hive.dart' show Hive, Box;
+import 'package:xayn_discovery_engine/src/domain/repository/engine_state_repo.dart'
+    show EngineStateRepository;
+import 'package:xayn_discovery_engine/src/infrastructure/box_name.dart'
+    show engineStateBox;
+
+/// Hive repository implementation of [EngineStateRepository].
+class HiveEngineStateRepository implements EngineStateRepository {
+  static const stateKey = 0;
+
+  Box<Uint8List> get box => Hive.box<Uint8List>(engineStateBox);
```

Comment:
This is the same approach that we use in other repo, but why we cannot open it only once in the constructor?

Reply:
constructors can't be async, so it would need to ba a static method + private default constructor. I think we could make this change for all repos, just not sure if this PR would be the best one to do it. Should we do that in a separate PR, wdyt? I can add a task for that.

Reply:
Yes we do this later. I thought there was some problem with hive in doing that.

Reply:
I saw that is not really opening anyway but only returning the handle to the open one so we can leave it like this.